Memory Exercise _ Yuyu

yuyudesigns opened the pull request.

yuyudesigns committed:
Memory Exercise _ Yuyu

afeld commented on index.html:
```diff
@@ -1,8 +1,95 @@
+  	margin:0px auto;
+  }
+  div#memory_board > div{
+  	background: url(tile_bg.jpg) no-repeat;
```
This file seems to be missing...

afeld commented on index.html:
```diff
@@ -1,8 +1,95 @@
+  var memory_values = [];
+  var memory_tile_ids = [];
+  var tiles_flipped = 0;
+  Array.prototype.memory_tile_shuffle = function(){
```
Modifying built-in classes is not a great idea...how else could you achieve this?

afeld commented on index.html:
```diff
@@ -1,8 +1,95 @@
+  var memory_array = ['A','A','B','B'];
+  var memory_values = [];
+  var memory_tile_ids = [];
+  var tiles_flipped = 0;
```
How could you avoid creating a bunch of global variables?

afeld commented on index.html:
```diff
@@ -1,8 +1,95 @@
+  			}
+  		}
+  	}
+  }
```
This is a very long method...how could you fix that?

afeld commented on index.html:
```diff
@@ -1,8 +1,95 @@
+  	var output = '';
+      memory_array.memory_tile_shuffle();
+  	for(var i = 0; i < memory_array.length; i++){
+  		output += '<div id="tile_'+i+'" onclick="memoryFlipTile(this,\''+memory_array[i]+'\')"></div>';
```
How can you avoid doing inline JS?

afeld replied:
In other words, how can we make it [unobtrusive](http://en.wikipedia.org/wiki/Unobtrusive_JavaScript)?

afeld commented on index.html:
```diff
@@ -1,8 +1,95 @@
+</head>
+<body>
+  <div id="memory_board"></div>
+  <script>newBoard();</script>
```
How could you avoid needing to put this in a separate `<script>` tag?

afeld commented:
Good so far! A couple general notes:
- Use `===` instead of `==`.
- When you have this much CSS and JS for a page, move them to external files.
- Careful with indentation!

Looking forward to seeing [V3](https://github.com/advanced-js/memory#v3)!

yuyudesigns committed:
Update index.html

Tried to play with another source code I found. except that I need to link the images, I think this works better.

afeld commented on index.html:
```diff
@@ -1,8 +1,92 @@
+	} 
+
+function displayBack(i) {
+	document.getElementById('t'+i).innerHTML = '<div onclick="disp('+i+');return false;"><img src="'+back+'" width="60" height="60" alt="back" \/><\/div>';
```
Try to avoid doing inline JS – use [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) instead!

http://en.wikipedia.org/wiki/Unobtrusive_JavaScript

afeld commented on index.html:
```diff
@@ -1,8 +1,92 @@
+'img6.gif','img7.gif','img8.gif','img9.gif','img10.gif','img11.gif',
+'img12.gif','img13.gif','img14.gif'];
+
+function randOrd(a, b){
```
Not sure what "ord" means...

afeld commented on index.html:
```diff
@@ -1,8 +1,92 @@
+	return (Math.round(Math.random())-0.5);
+	} 
+
+var im = []; 
```
Why not call this `images`?

afeld commented on index.html:
```diff
@@ -1,8 +1,92 @@
+	document.getElementById('t'+i).innerHTML = '<div onclick="disp('+i+');return false;"><img src="'+back+'" width="60" height="60" alt="back" \/><\/div>';
+	} 
+	
+var ch1, ch2, tmr, tno, tid, cid, cnt; 
```
Please use more descriptive variable names!

afeld commented on index.html:
```diff
@@ -1,8 +1,92 @@
+	
+var ch1, ch2, tmr, tno, tid, cid, cnt; 
+
+window.onload=start; 
```
Considered bad style to use a method before it's defined in the file, even if it happens to work because of hoisting :unamused:

afeld commented on index.html:
```diff
@@ -1,8 +1,92 @@
+window.onload=start; 
+
+function start() {
+	for (var i = 0; i <= 29 ;i++) displayBack(i);
```
What does this hard-coded number correspond to?

afeld commented:
Thanks for taking another pass. I think you could benefit from focusing on readability of your code. Try walking someone else through it, and if that's difficult, there's a good indication that it could be cleaned up to make it explain itself better.
